src/OpenColorIO/ops/lut1d/Lut1DOpGPU.cpp

```diff
@@ -261,29 +262,28 @@ void GetLut1DGPUShaderProgram(GpuShaderCreatorRcPtr & shaderCreator,
                 // At this point 'dep' contains the raw half
                 // Note: Raw halfs for NaN floats cannot be computed using
                 //       floating-point operations.
-                ss.newLine() << ss.float2Decl("retVal") << ";";
-                ss.newLine() << "retVal.y = floor(dep / " << float(width - 1) << ");";       // floor( dep / (width-1) ))
-                ss.newLine() << "retVal.x = dep - retVal.y * " << float(width - 1) << ";";   // dep - retVal.y * (width-1)
-
-                ss.newLine() << "retVal.x = (retVal.x + 0.5) / " << float(width) << ";";   // (retVal.x + 0.5) / width;
-                ss.newLine() << "retVal.y = (retVal.y + 0.5) / " << float(height) << ";";  // (retVal.x + 0.5) / height;
             }
             else
             {
                 // Need clamp() to protect against f outside [0,1] causing a bogus x value.
                 // clamp( f, 0., 1.) * (dim - 1)
                 ss.newLine() << "float dep = clamp(f, 0.0, 1.0) * " << float(length - 1) << ";";
+            }
 
-                ss.newLine() << ss.float2Decl("retVal") << ";";
-                // float(int( dep / (width-1) ))
-                ss.newLine() << "retVal.y = float(int(dep / " << float(width - 1) << "));";
-                // dep - retVal.y * (width-1)
-                ss.newLine() << "retVal.x = dep - retVal.y * " << float(width - 1) << ";";
+            ss.newLine() << ss.float2Decl("retVal") << ";";
 
-                // (retVal.x + 0.5) / width;
-                ss.newLine() << "retVal.x = (retVal.x + 0.5) / " << float(width) << ";";
-                // (retVal.x + 0.5) / height;
-                ss.newLine() << "retVal.y = (retVal.y + 0.5) / " << float(height) << ";";
+            if (height == 1)
+            {
+                ss.newLine() << "retVal.x = (dep + 0.5) / " << float(width) << ";"; // (dep + 0.5) / width;
+                ss.newLine() << "retVal.y = 0.5;";
+            }
+            else
+            {
+                ss.newLine() << "retVal.y = floor(dep / " << float(width - 1) << ");";       // floor( dep / (width-1) ))
```

Comment:
Both path should be able to `floor`, since it's equivalent to `float(int(..))` with non-negative values.